worker-sys/src/types/durable_object/state.rs

```diff
@@ -15,4 +15,21 @@ extern "C" {
 
     #[wasm_bindgen(method, js_name=waitUntil)]
     pub fn wait_until(this: &DurableObjectState, promise: &js_sys::Promise);
+
+    #[wasm_bindgen(method, js_name=acceptWebSocket)]
+    pub fn accept_websocket(this: &DurableObjectState, ws: &web_sys::WebSocket);
+
+    #[wasm_bindgen(method, js_name=acceptWebSocket)]
+    pub fn accept_websocket_with_tags(
+        this: &DurableObjectState,
+        ws: &web_sys::WebSocket,
+        tags: Vec<JsValue>,
+    );
+
+    #[wasm_bindgen(method, js_name=getWebSockets)]
+    pub fn get_websockets(this: &DurableObjectState) -> Vec<web_sys::WebSocket>;
```

Comment:
Just curious, why do there need to be two versions of `acceptWebSocket` and `getWebSockets`? The JS and C++ methods all provide an optional/maybe for `tags`, would an `Option` not work here?

Reply:
In my opinion (I am not the author), it's a style choice but this is probably the right one (in Rust anyway).

It's slightly kinder to the caller to have two functions, and probably more idiomatic in Rust. Passing a parameter `None` hurts the readability of the code, because it's not obvious at a glance what that parameter means.

The rest of this crate seems to mostly follow this style as well, e.g. `ObjectNamespace::unique_id`/`unique_id_with_jurisdiction`, `Storage::list`/`list_with_options`...

Reply:
Yeah it's an unfortunate impedance mismatch between function overloading in JS and the lack thereof in Rust. I'm just trying to match the style of the rest of the repo which manually "un-overloads" the functions as Eric pointed out.